src/families/stellar/api/horizon.ts

```diff
@@ -40,21 +49,46 @@ StellarSdk.HorizonAxiosClient.interceptors.response.use((response) => {
 });
 
 const getFormattedAmount = (amount: BigNumber) => {
-  return amount
-    .div(new BigNumber(10).pow(currency.units[0].magnitude))
-    .toString(10);
+  // TODO: ??? not sure why we need this conditional, but it is more consistent
+  // with it. Otherwise, some transactions would just freeze on broadcasting at
+  // the very end.
```

Comment:
Very weird, not something we are seeing currently with the existing code.
Would be to investigate a bit the broadcast failure to understand the link with this.

Reply:
Couldn’t reproduce this again, it must have been caused by work in progress.